bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java

```diff
@@ -154,22 +155,26 @@ public String getMapUrl(String vacuumMap, String country) throws MiCloudExceptio
         String mapResponse = request(url, map);
         logger.trace("response: {}", mapResponse);
         String errorMsg = "";
-        JsonElement response = PARSER.parse(mapResponse);
-        if (response.isJsonObject()) {
-            logger.debug("Received  JSON message {}", response.toString());
-            if (response.getAsJsonObject().has("result") && response.getAsJsonObject().get("result").isJsonObject()) {
-                JsonObject jo = response.getAsJsonObject().get("result").getAsJsonObject();
-                if (jo.has("url")) {
-                    String mapUrl = jo.get("url").getAsString();
-                    return mapUrl != null ? mapUrl : "";
+        try {
+            JsonElement response = PARSER.parse(mapResponse);
+            if (response.isJsonObject()) {
+                logger.debug("Received  JSON message {}", response);
+                if (response.getAsJsonObject().has("result")
+                        && response.getAsJsonObject().get("result").isJsonObject()) {
+                    JsonObject jo = response.getAsJsonObject().get("result").getAsJsonObject();
+                    if (jo.has("url")) {
+                        return jo.get("url").getAsString();
+                    } else {
+                        errorMsg = "Could not get url";
+                    }
                 } else {
-                    errorMsg = "Could not get url";
+                    errorMsg = "Could not get result";
                 }
             } else {
-                errorMsg = "Could not get result";
+                errorMsg = "Received message is invalid JSON";
             }
-        } else {
-            errorMsg = "Received message is invalid JSON";
+        } catch (ClassCastException | IllegalStateException e) {
```

Comment:
What would throw either of these exceptions?

Reply:
During my testing I had all sorts of garbage responses that came form the Xiaomi cloud. (so invalid json) 
Causing exceptions during the reading and processing of the json responses.
At most places I was already catching these, but seems this one not.